src/client/dfs/dfs.c

```diff
@@ -1754,6 +1805,182 @@ dfs_get_chunk_size(dfs_obj_t *obj, daos_size_t *chunk_size)
 	return 0;
 }
 
+void
+dfs_obj_copy_attr(dfs_obj_t *obj, dfs_obj_t *src_obj)
+{
+	if (S_ISDIR(obj->mode)) {
+		obj->d.oclass = src_obj->d.oclass;
+		obj->d.chunk_size = src_obj->d.chunk_size;
+	}
+}
+
+int
+dfs_obj_get_info(dfs_t *dfs, dfs_obj_t *obj, dfs_obj_info_t *info)
+{
+	int	rc;
+
+	if (obj == NULL || info == NULL)
+		return EINVAL;
+
+	switch (obj->mode & S_IFMT) {
+	case S_IFDIR:
+		if (obj->d.oclass)
+			info->doi_oclass_id = obj->d.oclass;
+		else if (dfs->attr.da_oclass_id)
+			info->doi_oclass_id = dfs->attr.da_oclass_id;
+		else
+			info->doi_oclass_id = daos_obj_get_oclass
+				(dfs->coh, 0, 0, 0);
+
+		if (obj->d.chunk_size)
+			info->doi_chunk_size = obj->d.chunk_size;
+		else if (dfs->attr.da_chunk_size)
+			info->doi_chunk_size = dfs->attr.da_chunk_size;
+		else
+			info->doi_chunk_size =  DFS_DEFAULT_CHUNK_SIZE;
+
+		break;
+	case S_IFREG:
+	{
+		daos_size_t cell_size;
+
+		rc = daos_array_get_attr(obj->oh, &info->doi_chunk_size,
+					 &cell_size);
+		if (rc)
+			return daos_der2errno(rc);
+
+		info->doi_oclass_id = daos_obj_id2class(obj->oid);
+		break;
+	}
+	case S_IFLNK:
+		info->doi_oclass_id = 0;
+		info->doi_chunk_size = 0;
+		break;
+	default:
+		D_ERROR("Invalid entry type (not a dir, file, symlink).\n");
+		return EINVAL;
+	}
+
+	return 0;
+}
+
+int
+dfs_obj_set_oclass(dfs_t *dfs, dfs_obj_t *obj, int flags, daos_oclass_id_t cid)
+{
+	daos_handle_t		oh;
+	d_sg_list_t		sgl;
+	d_iov_t			sg_iov;
+	daos_iod_t		iod;
+	daos_recx_t		recx;
+	daos_key_t		dkey;
+	int			rc;
+
+	if (obj == NULL)
+		return EINVAL;
+	if (!S_ISDIR(obj->mode))
+		return ENOTSUP;
+	if (cid == 0)
+		cid = dfs->attr.da_oclass_id;
+	/** 0 is default, allow setting it */
+	if (cid != 0 && !daos_oclass_is_valid(cid))
```

Comment:
Do you need the second zero check?
```suggestion
	/** 0 is default, allow setting it */
	if (cid == 0)
		cid = dfs->attr.da_oclass_id;
	if (!daos_oclass_is_valid(cid))
```

Reply:
dfs->attr.da_oclass_id can still be 0.